lib/src/preview/client.dart

```diff
@@ -19,38 +19,99 @@ class PreviewClient {
   final String _baseUrl;
   Network _network;
 
-  String _getPreviewUrl(String remotePath, int width, int height) {
-    return '$_baseUrl/index.php/core/preview.png?file=${Uri.encodeQueryComponent(remotePath)}&x=$width&y=$height&a=1&mode=cover&forceIcon=0';
-  }
+  /// fetch preview for [filePath] with provided [width] and [height]
+  Future<Uint8List> getPreviewByPath(String filePath, int width, int height,
+          {bool a = true,
+          String mode = 'cover',
+          bool forceIcon = false}) async =>
+      _getPreview(_createPreviewUrl(
+          filePath: filePath,
+          width: width,
+          height: height,
+          a: a,
+          mode: mode,
+          forceIcon: forceIcon));
 
-  String _getThumbnailUrl(String remotePath, int width, int height) {
-    if (!remotePath.startsWith('/')) {
-      remotePath = '/$remotePath';
-    }
-    return '$_baseUrl/index.php/apps/files/api/v1/thumbnail/$width/$height${Uri.encodeFull(remotePath)}';
-  }
+  /// fetch preview for [filePath] with provided [width] and [height] as ByteStream
+  Future<ByteStream> getPreviewStreamByPath(
```

Comment:
Do we actually need a api to return preview streams? Previews should not be that big. :thinking:

Reply:
I agree, but there might be a use for example in flutter using the `FutureBuilder` in a `ListView` or something like that